src/Query/TaskGroup.ts

```diff
@@ -68,6 +68,23 @@ export class TaskGroup {
         }
     }
 
+    /**
+     * Limits {@link tasks} array to a certain number. Tasks exceeding
+     * the limit will be removed from the end, shall be called on sorted tasks.
+     *
+     * @param limit number of tasks for the group to have. If greater
+     * than the task count, no action will be taken.
+     *
+     */
+    public applyTaskLimit(limit: number) {
+        if (limit >= this.tasks.length) {
```

Comment:
ok, I just thought I should keep the `TaskGroup.tasks` as `public readonly`. This requires to change it to `public` since `slice()` returns the new array and `splice` mutates it.

Reply:
> ok, I just thought I should keep the `TaskGroup.tasks` as `public readonly`. This requires to change it to `public` since `slice()` returns the new array and `splice` mutates it.

Interesting. Removing the readonly seems harmless.

Reply:
Agreed =) all the class' parameters were declared as `readonly` for some reason but I couldn't guess it.